### ISSUE TITLE
Style/SafeNavigation recommendation can introduce bugs

### ISSUE THREAD

**Issue by haines** [bbatsov/rubocop#5629]
For expressions of the form

```ruby
person && person.name.nil? # or .blank? if using ActiveSupport
```

the Style/SafeNavigation cop recommends

```ruby
person&.name.nil?
```

which is different and wrong. The former is checking "do we have a person without a name?" but the latter is checking "do we not have a person with a name?".

I think there should be an exception in this cop for expressions of this form.

--------

## Expected behavior

Style/SafeNavigation cop to not suggest fixing this violation.

## Actual behavior

Style/SafeNavigation suggests a fix that will introduce a bug.

## Steps to reproduce the problem

Lint the above code.

## RuboCop version

0.52.1 (using Parser 2.4.0.2, running on ruby 2.3.0 x86_64-darwin17)

**Comment by rrosenblum** [comment]
This issue is a duplicate of several other issues. The issue has been fixed in master. Please test against master and let me know if an issue still exists.

Fixed by #5606, #5595, and #5419.

**Comment by haines** [comment]
Looks good on master, thanks!
